src/workerd/api/tests/js-rpc-test.js

```diff
@@ -302,6 +302,38 @@ export class MyService extends WorkerEntrypoint {
   async roundTrip(value) {
     return value;
   }
+
+  async returnHeaders() {
+    let result = new Headers();
+    result.append("foo", "bar");
+    result.append("Set-Cookie", "abc");
+    result.append("set-cookie", "def");
+    result.append("corge", "!@#");
+    result.append("Content-Length", "123");
+    return result;
+  }
+
+  async returnRequest() {
+    return new Request("http://my-url.com", {
+      method: "PUT",
+      headers: {
+        "Accept-Encoding": "bazzip",
+        "Foo": "Bar"
+      },
+      redirect: "manual",
+      body: "Hello every body!",
+    });
+  }
```

Comment:
A test that verifies that a `Request` with an `AbortSignal` fails as expected would be a good idea.

Reply:
ec2b08c2